sentinelhub/api/catalog.py

```diff
@@ -265,3 +266,47 @@ def get_ids(self) -> List[str]:
         :return: A list of IDs
         """
         return [feature["id"] for feature in self]
+
+
+def get_available_timestamps(
+    bbox: BBox,
+    time_interval: Optional[RawTimeIntervalType],
+    data_collection: DataCollection,
+    *,
+    time_difference: Optional[dt.timedelta] = None,
+    ignore_tz: bool = True,
+    maxcc: Optional[float] = 1.0,
+    config: Optional[SHConfig] = None,
+) -> List[dt.datetime]:
+    """Helper function to search for all available timestamps for a given area and query parameters.
+
+    :param bbox: A bounding box of the search area.
+    :param data_collection: A collection specifying the satellite data source for finding available timestamps.
+    :param time_interval: A time interval from which to provide the timestamps.
+    :param time_difference: Shortest allowed time difference. Consecutive timestamps will be skipped if too close to
+        the previous one. Defaults to keeping all timestamps.
+    :param ignore_tz: Ignore the time zone part in the returned timestamps. Default is True.
+    :param maxcc: Maximum cloud coverage filter from interval [0, 1]. Default is None.
+    :param config: The SH configuration object.
+    :return: A list of timestamps of available observations.
+    """
+    query_filter = None
+    time_difference = time_difference if time_difference is not None else dt.timedelta(seconds=-1)
+    fields = {"include": ["properties.datetime"], "exclude": []}
+
+    if maxcc is not None and data_collection.has_cloud_coverage:
+        if isinstance(maxcc, (int, float)) and (maxcc < 0 or maxcc > 1):
+            raise ValueError('Maximum cloud coverage "maxcc" parameter should be a float on an interval [0, 1]')
+        query_filter = f"eo:cloud_cover < {int(maxcc * 100)}"
+
+    if data_collection.service_url is not None:
+        config = config.copy() if config else SHConfig()
+        config.sh_base_url = data_collection.service_url
+
```

Comment:
this is just so ugly :( but i guess we leave it for now

Reply:
does anything change with the new config version? I guess I could put the url in the config init, but it doesn't change much.

I wonder what would be the proper way to go about such things.

Reply:
signatures would need to be changed, 4.0 perhaps